### PR TITLE
Update troubleshooting guide for local disk costs.

### PR DESCRIPTION
**Pull request by thomasvn**
## Related issue #

Documents https://github.com/opencost/opencost/pull/2441

## Proposed Changes

As titled.

### CODE REVIEW

**Review by srpomeroy — DISMISSED** [review]

File: troubleshooting/troubleshoot-install.md [hunk 1]
```diff
@@ -399,4 +399,19 @@ gcpprovider.go Error loading metadata cluster-name: Get "http://169.254.169.254/
...
+Some cloud providers do not charge you for local disks physically attached to the node (i.e. ephemeral storage). By default, Kubecost monitors your local disk usage/capacity and applies an associated cost. To disable this feature use the following config:
+
+```yaml
+kubecostModel:
+  extraEnv:
+    - name: "ASSET_INCLUDE_LOCAL_DISK_COST"
+      value: "false"
+```
+
+This config will need to be applied to every cluster for which you do not want to charge for local disk cost. This config change only affects data moving forward. To fix historical data, you will need to [repair Asset & Allocation ETL](/troubleshooting/etl-repair.md) for each affected cluster, then wait for Kubecost's Aggregator to reingest the updated ETL data.
```
Comment (chipzoller) [inline troubleshooting/troubleshoot-install.md:413]:
```suggestion
This configuration will need to be applied to any cluster on which you do not want to charge for local disks and only takes effect on data moving forward. To fix historical data, you will need to [repair Asset & Allocation ETL](/troubleshooting/etl-repair.md) for each affected cluster, then wait for Kubecost's Aggregator to reingest the updated ETL data.
```

**Review by chipzoller — COMMENTED** [review]
I know this isn't exactly in scope for the changes you've made, but I suggest we add a paragraph to this page explaining to users why and when this procedure is necessary. What are the symptoms one may experience where such a procedure would be warranted/required? The closest I see is this last sentence before the first heading:

> In the event that you lose or are looking to rebuild the ETL data, the following endpoints should be used.

This doesn't tell me why I would need to rebuild ETL. Also, what does "endpoints" mean here?

File: troubleshooting/troubleshoot-install.md [hunk 2]
```diff
@@ -383,11 +383,11 @@ aws ecr-public get-login-password --region us-east-1 | helm registry login --use
 3. Change both entries to localhost:9001 and localhost:9003
 4. Restart the kubecost-cost-analyzer pod in the kubecost namespace
 
-### What is the difference between `.Values.kubecostToken` and `Values.kubecostProductConfigs.productKey`?
+### What is the difference between `kubecostToken` and `productKey`?
```
[the remark this answers is not in the document]
Comment (thomasvn) [reply troubleshooting/troubleshoot-install.md]: Fixed! Removed question mark.

File: troubleshooting/troubleshoot-install.md [hunk 3]
```diff
@@ -399,4 +399,19 @@ gcpprovider.go Error loading metadata cluster-name: Get "http://169.254.169.254/
 ```
 {% endcode %}
 
+### Local disks showing costs in Assets
+
+Some cloud providers do not charge you for local disks physically attached to the node (i.e. ephemeral storage). By default, Kubecost monitors your local disk usage/capacity and applies an associated cost. To disable this feature use the following config:
```
[the remark this answers is not in the document]
Comment (thomasvn) [reply troubleshooting/troubleshoot-install.md:404]: Ahh thank you! I always forget precisely what the it translates to in Latin. I previously always thought of "i.e." as in other words and "e.g." as for example ... which isn't _completely_ but also not completely accurate.

Yes you're right that in this instance "i.e. ephemeral storage" isn't technically accurate. I believe changing to "e.g." would correct this? Doing that now.

File: troubleshooting/troubleshoot-install.md [hunk 4]
```diff
@@ -399,4 +399,19 @@ gcpprovider.go Error loading metadata cluster-name: Get "http://169.254.169.254/
...
+
+```yaml
+kubecostModel:
+  extraEnv:
+    - name: "ASSET_INCLUDE_LOCAL_DISK_COST"
+      value: "false"
```
[the remark this answers is not in the document]
Comment (thomasvn) [reply troubleshooting/troubleshoot-install.md:410]: Noted! Here's the PR I've started. https://github.com/kubecost/cost-analyzer-helm-chart/pull/3449

**Comment by chipzoller** [comment]
Comments all resolved, you feel?

**Comment by thomasvn** [comment]
Yes I think so! Let me know if anything needs to be revisited.

File: troubleshooting/etl-repair.md [hunk 5]
```diff
@@ -1,8 +1,20 @@
...
+There are numerous reasons why you may need to repair ETL data. Please reach out to Kubecost Support for further guidance. The following conditions are cases in which repairing ETL data may help, but may not be the root cause of the issue:
+
+- If Kubecost is not showing data for a specific time window
+- If Kubecost's Assets data for a specific cluster is incorrect
+- If Kubecost's Allocation data for a specific cluster is incorrect
+
+In an enterprise multi-cluster environment, each individual Kubecost deployment builds its own ETL data before pushing it to the bucket. Therefore to repair data from an affected cluster, you must port-forward to that cluster's Kubecost deployment then run the repair command. After a repair has been completed on any cluster in your environment, the Kubecost Aggregator will detect that the ETL file has been recently modified, and ingest the new data into its database.
+
+{% hint style="warning" %}
+Remember that ETL files are computed based on Prometheus metrics. Before repairing ETL files, please confirm that your Prometheus server is still retaining metrics for the dates you are aiming to repair. `.Values.prometheus.server.retention`.
```
Comment (chipzoller) [inline troubleshooting/etl-repair.md:16]:
```suggestion
Remember that ETL files are computed based on Prometheus metrics. Before repairing ETL files, please confirm that your Prometheus server is still retaining metrics for the dates you are aiming to repair. Prometheus rentention is controlled by the Helm value `prometheus.server.retention`.
```

**Comment by chipzoller** [comment]
I think we should target `v2.3-content` for this and other PRs which are all 2.3 so we can send one PR to `main` to make docs live.

**Comment by thomasvn** [comment]
Ok I've updated the target branch from `main` to `v2.3-content`. The diff looks a little weird though. It seems to be pulling in additional changes I didn't make from the `main` branch to put into `v2.3-content`.

**Comment by chipzoller** [comment]
It's probably due to the merge from `main` in the PR. May just need to start again.

**Comment by thomasvn** [comment]
@chipzoller Yep, will do!

**Pull request closed without merging**